packages/expect/src/matchers.ts

```diff
@@ -134,16 +134,41 @@ const matchers: MatchersObject = {
     ensureNumbers(received, expected, matcherName, options);
 
     let pass = false;
-    let expectedDiff = 0;
-    let receivedDiff = 0;
-
-    if (received === Infinity && expected === Infinity) {
-      pass = true; // Infinity - Infinity is NaN
-    } else if (received === -Infinity && expected === -Infinity) {
-      pass = true; // -Infinity - -Infinity is NaN
+    let expectedDiff: number | bigint = 0;
+    let receivedDiff: number | bigint = 0;
+
+    //If both arent numbers, then one might be bigint
+    if (typeof received === 'number' && typeof expected === 'number') {
+      if (received === Infinity && expected === Infinity) {
+        pass = true; // Infinity - Infinity is NaN
+      } else if (received === -Infinity && expected === -Infinity) {
+        pass = true; // -Infinity - -Infinity is NaN
+      } else {
+        // Set default precision for numbers
+        precision = isNaN(precision) ? 2 : precision;
+        expectedDiff = Math.pow(10, -precision) / 2;
+        receivedDiff = Math.abs(expected - received);
+        pass = receivedDiff < expectedDiff;
+      }
     } else {
-      expectedDiff = Math.pow(10, -precision) / 2;
-      receivedDiff = Math.abs(expected - received);
+      // Set default precision for big integers
+      precision = isNaN(precision) ? -2 : precision;
+      if (precision >= 0) {
+        throw new Error(
+          matcherErrorMessage(
+            matcherHint(matcherName, undefined, undefined, options),
+            `${EXPECTED_COLOR(
+              'precision',
+            )} value must be a negative number for BigInts`,
+            printWithType('Precision', precision, printExpected),
+          ),
+        );
+      }
+      expectedDiff = BigInt(Math.pow(10, -precision) / 2);
```

Comment:
According to my limited understanding of proposal, I think we need to avoid casts as much as possible.

How about something like `(10n ** -precision) / 2n` if it is possible for exponent to be `number` otherwise `BigInt(-precision)` seems tolerable.

Reply:
@pedrottimark , exponent cant be a number,  Any bigint operations will be slower than numbers, so we may be better off doing the operations then casting. rather casting precision and doing the op?

Reply:
removed bigint support for tobeclosto